eds4jinja2/adapters/local_sparql_ds.py

```diff
@@ -29,18 +29,23 @@ def __init__(self, filename):
     def __reduce_bound_triple_to_string_format(self, dict_of_bound_variables: dict):
         return {str(k): str(v) for k, v in dict_of_bound_variables.items()}
 
-    def with_query(self, sparql_query: str) -> 'RDFFileDataSource':
+    def with_query(self, sparql_query: str, **kwargs) -> 'RDFFileDataSource':
         """
             Set the query text and return the reference to self for chaining.
         :return:
         """
         if self.__query__ != "":
             raise Exception("The query was already set.")
 
+        if kwargs:
```

Comment:
Cleaner and more pythonic would be to use the string Template: with a custom (or standard $) variable marker and pass the dict as a parameter to the safe_substitute(d) function. 

https://docs.python.org/3.8/library/string.html#template-strings